libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -140,20 +140,21 @@ string IRGeneratorForStatements::code() const
 
 void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _varDecl)
 {
-	solAssert(m_context.isStateVariable(_varDecl), "Must be a state variable.");
+	solAssert(_varDecl.immutable() || m_context.isStateVariable(_varDecl), "Must be immutable or a state variable.");
```

Comment:
Since it's an error to read immutables during creation as well as to not write to them explicitly during construction, do we really need/want to initialize them?

Reply:
This is codegen for `uint immutable x = 7;`

Reply:
Ah, right - on a second look it's fine then!